src/extension/app.tsx

```diff
@@ -43,14 +44,19 @@ chrome.runtime.sendMessage({ fetch: true }, function (response) {
             let tabId = entry.tabId;
             let id = json.envelope.impressionId;
             if (!(id in structured)) {
-                structured[id] = { envelope: json.envelope, events: [] };
-                structured[id]["envelope"].dateTime = entry.dateTime;
-                structured[id]["envelope"].summary = [];
+                if (json.envelope.sequenceNumber !== 0) {
+                    console.warn(`First payload for impression id ${id} has sequence number ${json.envelope.sequenceNumber}`);
+                    continue;
```

Comment:
Can we have better handling of errors like this. It should throw or indicate the data is corrupted rather than just continuing. That way the caller can trap errors and display an error message.

Reply:
Ideally, yes. However, currently unexpected events are piped to console.warn. Creating developed error handling should be a separate PR

---
In reply to: [144541577](https://github.com/Microsoft/clarity-tools/pull/23#discussion_r144541577) [](ancestors = 144541577)